React/Modules/RCTHapticFeedback.m

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTHapticFeedback.h"
+#import <UIKit/UIKit.h>
+
+@implementation RCTHapticFeedback
+
+{
+  UIImpactFeedbackGenerator *_impactFeedback;
+  UINotificationFeedbackGenerator *_notificationFeedback;
+  UISelectionFeedbackGenerator *_selectionFeedback;
+}
+@synthesize bridge = _bridge;
+
+RCT_EXPORT_MODULE()
+
+- (void)setBridge:(RCTBridge *)bridge
+{
+  _bridge = bridge;
+  if ([UIFeedbackGenerator class]) {
+    _impactFeedback = [UIImpactFeedbackGenerator new];
+    _notificationFeedback = [UINotificationFeedbackGenerator new];
+    _selectionFeedback = [UISelectionFeedbackGenerator new];
+  }
+}
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+RCT_EXPORT_METHOD(generate:(NSString *)type)
+{
+  if ([type isEqual: @"impact"]) {
+    [_impactFeedback impactOccurred];
+  } else if ([type isEqual:@"notification"]) {
+    [_notificationFeedback notificationOccurred:UINotificationFeedbackTypeWarning];
```

Comment:
and this has 3 more types that would be good to expose UINotificationFeedbackType.